tests/cache/models/samples/upsertBulk.js

```diff
@@ -290,6 +290,62 @@ describe('api::redisEnabled::POST::bulkUpsert ' + path, () => {
     });
   });
 
+  it('bulk upsert should return OK even if every sample in the request ' +
+    'has readonly fields in them', (done) => {
+    api.post(path)
+    .set('Authorization', token)
+    .send([
+      {
+        name: `${tu.namePrefix}Subject|${tu.namePrefix}AspectX`,
+        value: '2',
+        status: 'Invalid',
+      }, {
+        name: `${tu.namePrefix}Subject|${tu.namePrefix}AspectX`,
+        value: '4',
+        statusChangedAt: new Date().toString(),
+      },
+    ])
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        return done(err);
+      }
+      expect(res.body.status).to.equal('OK');
+      return done();
+    });
+  });
+
+  it('samples with read only fields in them should not be upserted', (done) => {
```

Comment:
No, just rejecting the ones with the read-only field in them. I can update the test to show that.